common/static_vector.h

```diff
@@ -0,0 +1,139 @@
+#pragma once
+#include <array>
+
+#include "utils.h"
```

Comment:
`utils.h`? What for?
Also don't need `array`, just `#include <algorithm>` is enough.